### ISSUE TITLE
Ctags package retrieving failed: Maximum file size exceeded

### ISSUE THREAD

**Issue by guoxiao** [msys2/MSYS2-packages#942]
```
$ pacman -S ctags
resolving dependencies...
looking for conflicting packages...

Packages (1) ctags-5.8-1

Total Download Size:   0.11 MiB
Total Installed Size:  0.26 MiB

:: Proceed with installation? [Y/n] y
:: Retrieving packages...
error: failed retrieving file 'ctags-5.8-1-x86_64.pkg.tar.xz' from repo.msys2.org : Maximum file size exceeded
error: failed retrieving file 'ctags-5.8-1-x86_64.pkg.tar.xz' from downloads.sourceforge.net : Maximum file size exceeded
error: failed retrieving file 'ctags-5.8-1-x86_64.pkg.tar.xz' from www2.futureware.at : Maximum file size exceeded
warning: failed to retrieve some files
error: failed to commit transaction (download library error)
Errors occurred, no packages were upgraded.
```

**Comment by sw-samuraj** [comment]
I have exactly the same error for `ctags`. Other packages install fine.

**Comment by stupidnetizen** [comment]
I am facing the same error. How to resolve this one?

**Comment by Qiguang** [comment]
I have that problem too

**Comment by ErichDonGubler** [comment]
What seems broken is the download of the `PKGBUILD` from SourceForge, because for some reason the new link is getting redirected. I've been able to get around this problem by downloading the `PKGBUILD` and installing manually using `makepkg`:

```bash
curl -O https://raw.githubusercontent.com/Alexpux/MSYS2-packages/master/ctags/PKGBUILD
makepkg -sri
```

**Comment by sw-samuraj** [comment]
Thanks @ErichDonGubler, it works!

In case, you are having those two errors:

```shell
==> ERROR: Cannot find the strip binary required for object file stripping.
```

or

```shell
==> ERROR: A failure occurred in build().
    Aborting...
````

you need to install build and compile packages:

```shell
pacman -S  binutils make gcc
```

**Comment by mati865** [comment]
@sw-samuraj it should be `pacman -S base-devel`.

**Comment by sp0ng3w0rthy** [comment]
Still getting the same issue even after attempting build from PKGBUILD after installing `binutils make gcc`.

**Comment by ErichDonGubler** [comment]
@sp0ng3w0rthy: You might want to try downloading the archive yourself and putting it into the same directory as the PKGBUILD -- `makepkg`, should detect that it's already there and just use it.

**Comment by sp0ng3w0rthy** [comment]
Thank you for the reply @ErichDonGubler, I'll test it the minute I get to work, but I believe the error I made is attempting to run the `makepkg` without first downloading the `ctags` archive into the same directory *facepalm*.

**Comment by sp0ng3w0rthy** [comment]
After multiple failed builds I **finally** realized my issue: I've been using [ConEmu](https://conemu.github.io/) and the [cygwin/msys connector](https://conemu.github.io/en/CygwinMsysConnector.html#cygwinmsys-terminal-connector) in order to run the MSYS2 shell.
As stated [here](https://conemu.github.io/en/CygwinMsysConnector.html#msys2-task), the connector  actually sets the `MSYSTEM` variable to `MINGW64` , as opposed to setting it to `MSYS` like the `C:\dev\msys64\msys2_shell.cmd` script does.

After reading [this](https://sourceforge.net/p/msys2/discussion/general/thread/dcf8f4d3/#9c02) comment on Sourceforge (by DanAaronGoldman), I realized I should be using the `msys2_shell.cmd` in order to build using `makepkg` **AND** to download the `ctags` archive to the same folder as the PKGBUILD file (as kindly stated by @ErichDonGubler ).

Managed to build perfectly, I hope this helps anyone using ConEmu and the cygwin/msys connector and not realizing why this doesn't work.

**Comment by guoxiao** [comment]
I found if I manually download `http://repo.msys2.org/msys/x86_64/ctags-5.8-1-x86_64.pkg.tar.xz` and use `pacman -U` to install, it works well.

It should be a bug in the file download part of pacman.

Related debug log (by adding `--debug` to pacman)
```
debug: url: http://repo.msys2.org/msys/x86_64/ctags-5.8-1-x86_64.pkg.tar.xz
debug: maxsize: 113924
debug: opened tempfile for download: /var/cache/pacman/pkg/ctags-5.8-1-x86_64.pkg.tar.xz.part (wb)
debug: curl returned error 63 from transfer
error: failed retrieving file 'ctags-5.8-1-x86_64.pkg.tar.xz' from repo.msys2.org : Maximum file size exceeded
```
Don't know where the maxsize come from.

**Comment by cbrt64** [comment]
For whatever reason, the `%CSIZE%` given in /var/lib/pacman/sync/[msys.db](http://repo.msys2.org/msys/x86_64/msys.db)//ctags/desc for `ctags-5.8-1-x86_64.pkg.tar.xz` is wrong (113924 bytes instead of 126936). Similar discrepancies are affecting some other packages as well, as seen in the other duplicate issues. (ucl just got rebuilt yesterday, fixing this problem for that package.)

`pacman -U http://repo.msys2.org/msys/x86_64/ctags-5.8-1-x86_64.pkg.tar.xz` is the easiest workaround i know of, it even downloads the package into pacman's cache.

#1025, #1052, #1094 are definitely duplicates of this, #924 is related.

**Comment by earnie** [comment]
Coming into this quite late (ended up here because of google) but I don't see any mention of needing to ensure the DB is up-to-date via `pacman -Sy` before attempting the install with `pacman -S ctags`.  And to upgrade you should use `pacman -Syu` to update all packages.

Now obviously something else was amiss and as a result we have [#1109] but that doesn't negate my caution in the previous paragraph.

**Comment by guoxiao** [comment]
Should be fixed in #1109. Closing.
